webapp/channels/src/components/admin_console/access_control/editors/table_editor/table_editor.tsx

```diff
@@ -23,53 +26,36 @@ interface TableEditorProps {
     onChange: (value: string) => void;
     onValidate?: (isValid: boolean) => void;
     disabled?: boolean;
-    userAttributes: Array<{
-        attribute: string;
-        values: string[];
-    }>;
+    userAttributes: UserPropertyField[];
+    onParseError: (error: string) => void;
+    actions: {
+        getVisualAST: (expr: string) => Promise<ActionResult>;
+    };
 }
 
-// Parse CEL expression into table rows
-const parseExpression = async (expr: string): Promise<TableRow[]> => {
+// Parses a CEL (Common Expression Language) string into a structured array of TableRow objects.
+// This allows the expression to be displayed and edited in a user-friendly table format.
+export const parseExpression = (visualAST: AccessControlVisualAST): TableRow[] => {
```

Comment:
> A few things:
>
>   1. Would it be worth unit testing any of this? I know the server is handling most of this parsing, but this sort of logic tends to be complex enough to need that
>   2. The await on the client method and the places where we throw errors should be changed to return them. I think uncaught errors inside of a useEffect might crash the web app.
>   3. It's a bit boilerplatey, but could you convert this into a Redux action? Even if it doesn't use Redux directly, we don't typically call client methods from outside of actions.


https://github.com/mattermost/mattermost/pull/30785#discussion_r2089023150 comment from @hmhealey

Added test case, created a redux action and tried to handle exceptions in the `useEffect` in `L:101`